### PR TITLE
Remove leftover usage of the attrs library in runtime

### LINKED ISSUE #5806: [TODO] Finish wiping the attrs usage
### Describe the bug

Originally, @asvetlov intended to delete any use of `attrs` from aiohttp 4.0+ series via https://github.com/aio-libs/aiohttp/pull/5284. He dropped it from the runtime deps too.

But there are some places that still rely on `attrs` meaning that this will cause failures in runtime because of the missing dep. The tests currently succeed because attrs is listed as a test dep.

One place where it's used is https://github.com/aio-libs/aiohttp/blob/4d58400477331704132cc9427d66699e5815ee25/aiohttp/web_protocol.py#L23.

### To Reproduce

pip install the recent master w/o the test deps and it should fail

### Expected behavior

aiohttp shouldn't rely on deps not declared in the metadata.

### Logs/tracebacks

```python-traceback
N/A
```

### Python Version

```console
$ python --version
```

### aiohttp Version

```console
$ python -m pip show aiohttp
```

### multidict Version

```console
$ python -m pip show multidict
```

### yarl Version

```console
$ python -m pip show yarl
```

### OS

N/A

### Related component

Server

### Additional context

_No response_

### Code of Conduct

- [X] I agree to follow the aio-libs Code of Conduct

### PR DESCRIPTION
**Pull request by RawatDevanshu**
fix #5806

### CODE REVIEW

File: aiohttp/web_protocol.py [hunk 1]
```diff
@@ -19,8 +19,7 @@
     Union,
     cast,
 )
-
-import attr
+import dataclasses
```
Comment (webknjaz) [inline aiohttp/web_protocol.py:22]: I'd move this after the line 2 for consistency.

File: aiohttp/web_protocol.py [hunk 2]
```diff
@@ -105,7 +104,7 @@ async def log(
         self.access_logger.log(request, response, self._loop.time() - request_start)
 
 
-@attr.s(auto_attribs=True, frozen=True, slots=True)
+@dataclasses.dataclass(frozen =True)
```
Comment (webknjaz) [inline aiohttp/web_protocol.py]: Having a whitespace before `=` is a style violation. The linter would make the CI red.
```suggestion
@dataclasses.dataclass(frozen=True)
```

File: requirements/lint.txt [hunk 3]
```diff
@@ -8,10 +8,6 @@ appdirs==1.4.4
     # via
     #   black
     #   virtualenv
-attrs==20.3.0
-    # via
-    #   flake8-pyi
-    #   pytest
```
Comment (webknjaz) [inline requirements/lint.txt:14]: Don't edit this file, it's managed automatically. In this case, `attrs` comes in as a transitive dependency as visible in the comment.

File: requirements/dev.txt [hunk 4]
```diff
@@ -19,11 +19,6 @@ appdirs==1.4.4
     #   virtualenv
 async-timeout==4.0.0a3
     # via -r requirements/base.txt
-attrs==20.3.0
-    # via
-    #   -r requirements/lint.txt
-    #   flake8-pyi
-    #   pytest
```
Comment (webknjaz) [inline requirements/dev.txt:26]: Don't edit this file, it's managed automatically. In this case, `attrs` comes in as a transitive dependency as visible in the comment.

**Comment by webknjaz** [comment]
Additionally to the inline comments, you need to drop https://github.com/aio-libs/aiohttp/blob/fd2ea56/tests/autobahn/Dockerfile.aiohttp#L8-L17.

File: aiohttp/web_protocol.py [hunk 5]
```diff
@@ -19,8 +19,7 @@
     Union,
     cast,
 )
-
-import attr
+import dataclasses
 import yarl
```
Comment (webknjaz) [inline aiohttp/web_protocol.py:24]: Recover the empty line that separates stdlib and third-party sections of imports
```suggestion

import yarl
```

**Comment by Dreamsorcerer** [comment]
Hi, any interest in finishing this PR? Appears there are 2 minor changes needed (moving the import and reverting the changes to requirements files).

**Comment by Dreamsorcerer** [comment]
Or enable the option to "Allow commits from maintainer" so I can finish it...?

**Comment by webknjaz** [comment]
> Or enable the option to "Allow commits from maintainer" so I can finish it...?

You can still do this by fetching their branch and pushing to your fork.

http://blog.rolpdog.com/2018/08/testing-and-modifying-github-prs.html

**Comment by Dreamsorcerer** [comment]
Well, they've actually deleted their repo, so looks like I'd need to copy manually from their diff and figure out how to attribute them in Git.

I think it's probably faster for me to redo it from scratch.

**Comment by Dreamsorcerer** [comment]
Replaced by #5960.

**Comment by webknjaz** [comment]
_(educational comment)_ @Dreamsorcerer FYI this is how you can fetch *read-only* fork branches: `git config --add remote.origin.fetch '+refs/pull/*/head:refs/pull/origin/*'`. Provided that your `origin` points to this upstream repo, that is, change it if it isn't. Having done this once, it'll keep the PR branches up-to-date in your local repo with every `git fetch`.

**Pull request closed without merging**